lib/src/extensions.dart

```diff
@@ -148,6 +148,44 @@ extension AstNodeExtension on AstNode? {
     }
     return null;
   }
+
+  /// Builds the list resulting from traversing the node in DFS and does not
+  /// include the node itself.
+  ///
+  /// It excludes the nodes for which the [excludeCriteria] returns true. If
+  /// [excludeCriteria] is not provided, all nodes are included.
+  Iterable<AstNode> traverseNodesInDFS({AstNodePredicate? excludeCriteria}) {
+    var self = this;
+    if (self == null) {
+      return [];
+    }
+    var nodes = <AstNode>{};
+    var nodesToVisit = List.of(self.childEntities.whereType<AstNode>());
```

Comment:
Also not necessary.

Consider adding a private extension method on `AstNode` like
```
List<AstNode> get childNodes => self.childEntities.whereType<AstNode>();
```

Reply:
Good call, done.